src/components/ui/Image/Picture.tsx

```diff
@@ -0,0 +1,50 @@
+import { Fragment, memo } from 'react'
+import { Helmet } from 'react-helmet-async'
+import type { HTMLAttributes } from 'react'
+
+import { useSources } from './useSources'
+import type { SourceOptions } from './useSources'
+import type { ImageOptions } from './useImage'
+
+interface Props extends HTMLAttributes<HTMLPictureElement> {
+  sources: SourceOptions[]
+  img: ImageOptions
+  preload?: boolean
+}
+
+function Picture({
+  sources: sourceOptions,
+  img: imgProps,
+  preload = false,
+  ...otherProps
+}: Props) {
+  const sources = useSources(sourceOptions)
+
+  return (
+    <picture>
+      {sources.map(({ srcSet, media, height, preloadLinks }, index) => {
+        return (
+          <Fragment key={index}>
```

Comment:
I'm wondering if using the index as a key (here and in the line 41) won't cause any issues (as exemplified [here](https://robinpokorny.com/blog/index-as-a-key-is-an-anti-pattern/)).

Reply:
Well noticed, Sara.

I think this picture is the [Exception from the rule](https://robinpokorny.com/blog/index-as-a-key-is-an-anti-pattern/#:~:text=Exception%20from%20the%20rule), though. Do you notice any case where the `sources` can change, reordered, or filtered?